06_puppy/lexual/puppy.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+var out io.Writer = os.Stdout
+
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+// Storer defines standard CRUD operations for Puppys.
+type Storer interface {
+	CreatePuppy(*Puppy)
```

Comment:
Might also be a good idea to return something from this `CreatPuppy` function. Otherwise how are you able to determine whether the call actually worked? Up to you, but you might want to return a bool like you did for `DeletePuppy` or maybe the created `Puppy`